src/Moq/Mock.cs

```diff
@@ -431,6 +431,8 @@ internal static void VerifyNoOtherCalls(Mock mock)
 
 		#region Setup
 
+		internal bool HasEventSetup { get; private set; } = false;
```

Comment:
`Mock` shouldn't have any state. This should be declared `abstract`, and then overridden in both `Mock<T>` and `AsInterface<TInterface>`.

Reply:
OK, I can change it like you asked.

Just as a side note - notice that this property is used only to preserve the optimization documented in a comment in InterceptionAspects.cs:
> Special case for event accessors. The following, seemingly random character checks are guards against more expensive checks (for the common case where the invoked method is *not* an event accessor).

We can eliminate this property at all if you prefer.

Reply:
Hmm, then I misunderstood what the new property is for. I thought it was needed to avoid / mitigate a breaking change (see discussion in preceding review comment). Will look into this more closely soon.

Don't worry too much about that particular micro-optimization. We can always optimize again at a later time (after code changes, if needed at all).